pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+                    status: ResponseStatus::Successed,
+                },
+            );
+
+            Self::deposit_event(Event::<T>::SlashRecorded(agent, amount));
+            Ok(().into())
+        }
+
+        // bond/unbond/rebond/bond_extra may be merge into one
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn record_bond_response(
+            origin: OriginFor<T>,
+            era_index: Option<EraIndex>,
+        ) -> DispatchResultWithPostInfo {
+            let _who = ensure_signed(origin)?;
+            let era_index = era_index.unwrap_or(Self::current_era());
+
+            let op = StakingOperationHistory::<T>::try_mutate(
+                era_index,
+                StakingOperationType::Bond,
+                |op| -> Result<Operation<_>, DispatchError> {
+                    let next_op = op
+                        .clone()
+                        .filter(|op| op.status == ResponseStatus::Ready)
+                        .map(|op| Operation {
+                            status: ResponseStatus::Successed,
+                            ..op
+                        })
+                        .ok_or(Error::<T>::OperationNotReady)?;
+                    *op = Some(next_op.clone());
+                    Ok(next_op)
+                },
+            )?;
+            T::Currency::deposit(T::LiquidCurrency::get(), &Self::account_id(), op.amount)?;
+            Self::deposit_event(Event::<T>::BondSucceed(era_index));
+            Ok(().into())
+        }
+
+        /// Invoked when unbonding extrinsic finished. Burn previously transfered xksm if
+        /// successed.
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn record_unbond_response(
+            origin: OriginFor<T>,
+            era_index: Option<EraIndex>,
+        ) -> DispatchResultWithPostInfo {
+            let _who = ensure_signed(origin)?;
+            let era_index = era_index.unwrap_or(Self::current_era());
+
+            let op = StakingOperationHistory::<T>::try_mutate(
+                &era_index,
+                StakingOperationType::Unbond,
+                |op| -> Result<Operation<_>, DispatchError> {
+                    let next_op = op
+                        .clone()
+                        .filter(|op| op.status == ResponseStatus::Ready)
+                        .map(|op| Operation {
+                            status: ResponseStatus::Successed,
+                            ..op
+                        })
+                        .ok_or(Error::<T>::OperationNotReady)?;
+                    *op = Some(next_op.clone());
+                    Ok(next_op)
+                },
+            )?;
+            T::Currency::withdraw(T::LiquidCurrency::get(), &Self::account_id(), op.amount)?;
+            Self::deposit_event(Event::<T>::UnbondSucceed(era_index));
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000)]
+        #[transactional]
```

Comment:
this function alone is useless now because we can already call xTokens to do this

Reply:
we should transfer assets from liquidStaking pallet, so the method is necessary

Reply:
we should transfer assets from liquidStaking pallet, so the method is necessary

Reply:
you are right